pyccel/codegen/printing/ccode.py

```diff
@@ -689,7 +691,7 @@ def _print_Allocate(self, expr):
         shape = ", ".join(a for a in shape)
         dtype = self._print(expr.variable.dtype)
         dtype = self.find_in_ndarray_type_registry(dtype, expr.variable.precision)
-        shape_Assign = "(int[]){" + shape + "}"
+        shape_Assign = "(int32_t[]){" + shape + "}"
```

Comment:
It would be better if you collected `int` using `find_in_dtype_registry` that way it would have been automatically updated